### PR TITLE
[E-ivan < T396] Implement setKey function on maps

### PR DESCRIPTION
**Pull request by imilinovic**
### Description

Implement setKey

### Pull request type

- [ ] Bugfix
- [X] Algorithm/Module
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

######################################

#### Module/Algorithm
- [ ] Core algorithm/module implementation
- [ ] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Unit tests
- [ ] End-to-end tests
- [ ] Code documentation
- [ ] README short description
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)
- [ ] Update GQLALchemy signatures in [query builder](https://github.com/memgraph/gqlalchemy/blob/main/gqlalchemy/graph_algorithms/query_builder.py  ) using [query module signature generator](https://github.com/memgraph/gqlalchemy/blob/main/scripts/query_module_signature_generator.py)

######################################

### CODE REVIEW

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Few small changes

File: cpp/map_module/algorithm/map.cpp [hunk 1]
```diff
@@ -1 +1,21 @@
...
+
+void Map::SetKey(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+
+  try {
+    auto map = arguments[0].ValueMap();
+    const auto &key{arguments[1].ValueString()};
+    const auto &value = arguments[2];
```
[the remark this answers is not in the document]
Comment (imilinovic) [reply cpp/map_module/algorithm/map.cpp]: Just a general question,
when is one style prefered over the other
`const auto variable{argument}`
vs
`const auto variable = argument`

File: cpp/map_module/algorithm/map.hpp [hunk 2]
```diff
@@ -1,5 +1,18 @@
 #pragma once
 
 #include <mgp.hpp>
+#include <string_view>
 
-namespace Map {}  // namespace Map
+namespace Map {
+
+constexpr const std::string_view kProcedureSetKey = "setKey";
```
Comment (antoniofilipovic) [inline cpp/map_module/algorithm/map.hpp:8]: I think it makes sesne to call this function `set_key` as this will be consistent with rest of our functions

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Small change and should be okay

File: cpp/map_module/algorithm/map.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (antoniofilipovic) [reply cpp/map_module/algorithm/map.cpp]: I am not sure, I saw that lot of things in our Memgraph codebase are constructed as `{}`, I tried to ask you guys on PRs to use that style but forget sometimes.

**Review by antoniofilipovic — APPROVED** [review]
Looks good to go

**Pull request merged**